### ISSUE TITLE
Gnome Shell 44 support

### ISSUE THREAD

**Issue by oae** [velitasali/GTKTitleBar#36]
The extension stopped working after the gnome-shell 44 upgrade. There are no errors in the logs also.

**Comment by oae** [comment]
Changing `_HIDE_FLAGS` to `['0x2', '0x0', '0x0', '0x0', '0x0'] ` 
https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L18

and the following line to `return hints[2] != '0x0';` works. But I don't know if this breaks the older versions of gnome-shell.

https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L62

**Comment by velitasali** [comment]
Thanks for investigating. I haven't checked the changes you suggested yet, but I guess we can remove the support for older versions if necessary. That said, I am experiencing some issues with GNOME 44 right now (like the one with the window focus), so I think it is better to wait a little bit more before adding the support for 44.

**Comment by OlegKrikun** [comment]
The proposed changes work for me on Gnome 44.rc

**Comment by oae** [comment]
I am experiencing window focus problems also.

**Comment by Hole-code** [comment]
changing metadata.json didn't solve the problem. The extension starts but does not work. Headlines remain

**Comment by Hole-code** [comment]
fedora 38 gnome 44

**Comment by rivenirvana** [comment]
Shell 44 version not on EGO yet.

**Comment by wanlce** [comment]
ArchLinux gnome44.1 X11
The gtktitlebar has no effect and cannot hide the title bar of any application.
![Peek 2023-05-09 13-25](https://user-images.githubusercontent.com/55907021/237001874-611bf74a-687f-490e-98ec-b3c6a64d6b82.gif)

**Comment by shmu26** [comment]
Same issue on Ubuntu 23.04

**Comment by wanlce** [comment]
> Changing `_HIDE_FLAGS` to `['0x2', '0x0', '0x0', '0x0', '0x0'] `
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L18
> 
> and the following line to `return hints[2] != '0x0';` works. But I don't know if this breaks the older versions of gnome-shell.
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L62

Thanks a lot, these changes make the extension work on Gnome44

**Comment by wojciech-zurek** [comment]
@oae  @wanlce Thank you. It works for me.

**Comment by Racsuu** [comment]
hi, not sure if this can help but i remember that gnome 44 manage on another way the title bars, it broke some themes like adw-gtk3 so maybe it can be related?
https://github.com/lassekongo83/adw-gtk3/issues/222
https://gitlab.gnome.org/GNOME/mutter/-/commit/20e2adc4fc0549a394a18cf207ad32b42b142d4f

**Comment by feimosi** [comment]
@velitasali any chance to release an update with the suggested changes? That'd be great, thank you

**Comment by jluquer** [comment]
> Changing `_HIDE_FLAGS` to `['0x2', '0x0', '0x0', '0x0', '0x0'] `
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L18
> 
> and the following line to `return hints[2] != '0x0';` works. But I don't know if this breaks the older versions of gnome-shell.
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L62

Thanks!! It worked for me. I found other bug, I don't know if it was happening before.

The bug I found is that when I lock the screen and go back the VSCode goes from full-screen to normal window mode.

I would be glad to help in any way. Thank you!!

**Comment by rafal06** [comment]
> > Changing `_HIDE_FLAGS` to `['0x2', '0x0', '0x0', '0x0', '0x0'] `
> > https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L18
> > 
> > and the following line to `return hints[2] != '0x0';` works. But I don't know if this breaks the older versions of gnome-shell.
> > https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L62
> 
> Thanks!! It worked for me. I found other bug, I don't know if it was happening before.
> 
> The bug I found is that when I lock the screen and go back the VSCode goes from full-screen to normal window mode.
> 
> I would be glad to help in any way. Thank you!!

I have this issue too

**Comment by konrad-garus** [comment]
The changes from @oae don't seem to work for me on Ubuntu 23.04.

**Comment by intelfx** [comment]
> Changing `_HIDE_FLAGS` to `['0x2', '0x0', '0x0', '0x0', '0x0'] `
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L18
> 
> and the following line to `return hints[2] != '0x0';` works. But I don't know if this breaks the older versions of gnome-shell.
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L62

This sort of works for me on Arch + GNOME 44.1, but 1) windows that open maximized do not have their title bar hidden (it only disappears after de-maximizing and re-maximizing the window) and 2) it breaks the JetBrains IDEs (their window menus instantly disappear when opened when the title bar is hidden, suggesting that something is wrong with the focus).

**Comment by intelfx** [comment]
Yup, doing a `focus(1)` on the window object after undecorating "fixes" (or, rather, works around) the issue. Any ideas on how to wait for the window to get undecorated programmatically?

**Comment by crillon** [comment]
> Changing `_HIDE_FLAGS` to `['0x2', '0x0', '0x0', '0x0', '0x0'] `
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L18
> 
> and the following line to `return hints[2] != '0x0';` works. But I don't know if this breaks the older versions of gnome-shell.
> 
> https://github.com/velitasali/gtktitlebar/blob/88c87e60e4b26328f951252ee84e498e4216eb6b/gtktitlebar%40velitasali.github.io/window.js#L62

Yep, this kinda works, but it didn't remove the titlebar if the window is **tiled** using a Material Shell extension (if the `Hide when` setting is set to `Both` or `Tiled`). This used to work in Gnome 43, but now it doesn't :(

**Comment by peter-lyons-kehl** [comment]
Thank you for GTK Title Bar.

Confirming that it stopped working (with no apparent error) in GNOME 44.1 that comes with Manjaro x64 X11 as of around June 5th, 2023.

To save you time, FYI: An alternative/similar extension https://github.com/alecdotninja/no-titlebar-when-maximized partially works with GNOME 44.1, but it "freezes" Terminal imput: https://github.com/alecdotninja/no-titlebar-when-maximized/issues/13.
